orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/config/SelectableService.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.config;
+
+import org.springframework.util.Assert;
+
+import java.util.List;
+
+public class SelectableService {
+  private final List<ServiceSelector> serviceSelectors;
+
+  public SelectableService(List<ServiceSelector> serviceSelectors) {
+    this.serviceSelectors = serviceSelectors;
+  }
+
+  public Object getService(Criteria criteria) {
+    Assert.notNull(criteria);
+
+    // look for a service selector that explicit supports this application (cannot be the default service selector)
+    Criteria applicationCriteria = new Criteria(
```

Comment:
from the discussion in slack, the prioritization could be achieved also by listing the different replicas in priority order with their matching rules and picking the first match